File: ejercicios/src/main/java/co/edu/sena/ejercicios/ejercicio05/Macho.java
package co.edu.sena.ejercicios.ejercicio05;

public class Macho extends Hormiga implements Obrero, Soldado, Explorador {
    @Override
    public void construir(){
        System.out.println("construyo construyo");
    }

    @Override
    public void arreglar(){
        System.out.println("arreglo los túneles del nido");
    }

    @Override
    public void recolectar() {
        System.out.println("recolecto comidita");
    }

    @Override
    public void excavar() {
        System.out.println("abro un agujero");
    }

    @Override
    public void cuidarCrias() {
        System.out.println("cuido a la prole");
    }

    @Override
    public void investigar() {
        System.out.println("salgo a investigar que hay alrededor");
    }

    @Override
    public void marcarCaminos() {
        System.out.println("marco los caminos para que mis compañeras no se pierdan");
    }

    @Override
    public void informar() {
        System.out.println("informo cuando encuentro algo de comer");
    }

    @Override
    public void luchar() {
        System.out.println("lucho lucho por mi colonia");
    }

    @Override
    public void defender() {
        System.out.println("defiendo mi colonia");
    }

    @Override
    public void vigilar() {
        System.out.println("vigilo que no entren intrusos a mi colonia");
    }
}

File: typescript/src/ejercicio05/macho.ts
import {Explorador} from "./explorador.interface";
import {Hormiga} from "./hormiga";
import {Obrero} from "./obrero.interface";
import {Soldado} from "./soldado.interface";


export class Macho extends Hormiga implements Soldado, Obrero, Explorador {
    public arreglar(): void {
        console.log("arreglo las cosas");
    }

    public construir(): void {
        console.log("construyo el nido");
    }

    public cuidarCrias(): void {
        console.log("cuido las crias");
    }

    public defender(): void {
        console.log("defiendo el nido");
    }

    public excavar(): void {
        console.log("excavo");
    }

    public informar(): void {
        console.log("informo");
    }

    public investigar(): void {
        console.log("investigo");
    }

    public luchar(): void {
        console.log("lucho");
    }

    public marcarCaminos(): void {
        console.log("marco el camino");
    }

    public recolectar(): void {
        console.log("recolecto alimentos");
    }

    public vigilar(): void {
        console.log("vigilo");
    }
}